core/src/modules/tableSpeculation.ts

```diff
@@ -57,4 +59,17 @@ export namespace TableSpeculation {
 
     return databaseType;
   }
+
+  export function suggestKey(
+    key: string,
+    model: GrouparooModel,
+    existingProperties: Property[]
+  ): string {
+    for (const property of existingProperties) {
+      if (property.key === key) {
+        return `${ConfigWriter.generateId(model.name)}_${key}`;
```

Comment:
Nice addition! This ensures that the name suggested is human-friendly, even before we safe the config files for something new in ui-config.